foundation/foundation-mda/src/main/resources/templates/data-delivery-spark/asynchronous.processor.base.java.vm

```diff
@@ -137,7 +137,7 @@ public abstract class ${step.capitalizedName}Base extends AbstractPipelineStep {
     #end
 #end
 
-#if ($step.hasMessagingOutbound() && $step.hasNativeInbound())
+#if ($step.hasMessagingOutbound() && ($step.hasNativeInbound() || $step.hasVoidInbound()))
```

Comment:
A: Just to help with maintenance, we should pull this boolean term out into a variable like `useEmitter` so that if we need to change this calculation again we only do it in one place.